homeassistant/components/sipcall/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""The SIP Call integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_HASS_CONFIG, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.NOTIFY]
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Discord component."""
+
+    hass.data[DATA_HASS_CONFIG] = config
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SIP Call from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    # We simply store the config data. The actual connection will be made by notify
+    hass.data[DOMAIN][entry.entry_id] = entry.data
+
```

Comment:
I don't know how SIP works, but is it maybe possible to check beforehand if we can connect to the server? To check if our credentials are still valid and we know the server is online and reachable? This can be very nice in both the init and in the config flow.

Reply:
Well... It would be doable if my nanosip implementation also supported the REGISTER method which is used to receive calls. But the aim of all this is to make it as simple as possible. 
So we can only verify the credentials by making a call, and I guess we don't want to do that during the config flow.

Reply:
I mean yes we like simple, but we love this as well. If you can guard people from setting up a config entry without ever testing it the data is correct, that'd be the best. Otherwise you're getting issues like "yea I have this entry setup and my device doesn't appear".

The config flow is the most user friendliest way of saying to the user they f*cked up and that it's their problem

Reply:
For this, I will need to make some changes to `nanosip`. It still won't support `REGISTER`, but the config flow can allow the user to make a test call.

Reply:
As you're the one knowing a lot about this, what would your ideal way of checking this be? You said about register, what does that do and would that work here? Why/why not?

A test call would be nice, it's something. Sadly we still can't verify if it worked ourselves

Reply:
Ok, it's done via a dummy test call to a recipient that does not exist. If the credentials are wrong, it will fail with a corresponding authentication error. Otherwise with a 404 error. This is used in the verification of the config flow.